static/js/components/EmploymentTab.js

```diff
@@ -65,56 +79,232 @@ class EmploymentTab extends ProfileTab {
     );
   }
 
-  employmentEntriesForm (index) {
-    let keySet = (key) => ['work_history', index, key];
+  componentWillMount () {
+    const { ui, setWorkHistoryEdit } = this.props;
+    if ( _.isUndefined(ui.workHistoryEdit) ) {
+      setWorkHistoryEdit(true);
+    }
+  }
+
+  toggleWorkHistoryEdit = () => {
+    const { ui, setWorkHistoryEdit } = this.props;
+    setWorkHistoryEdit(!ui.workHistoryEdit);
+  }
+
+  closeWorkDialog = () => {
+    const { setWorkDialogVisibility } = this.props;
+    setWorkDialogVisibility(false);
+  }
+
+  addWorkHistoryEntry = () => {
+    const {
+      updateProfile,
+      profile,
+      setWorkDialogIndex,
+      setWorkDialogVisibility,
+    } = this.props;
+    let clone = Object.assign({}, profile);
+    clone['work_history'] = clone['work_history'].concat(
+      _.cloneDeep(this.blankWorkHistoryEntry)
```

Comment:
On a side note, we should revert the state of the edit when the user clicks 'Cancel'. Since the user shouldn't have any state which is being edited which is not in the dialog box, it should be enough to clear the `edit` using the action similar to how `EducationTab` does it.